export/mlab_export_test.py

```diff
@@ -0,0 +1,492 @@
+            fake_filename, value_name, mock_options))
+    
+    self.assertEqual(returned_host, expected_host)
+    self.assertEqual(returned_experiment, expected_experiment)
+    self.assertEqual(returned_metric, expected_metric)
+    self.assertTrue(mock_logging.called)
+
+  @mock.patch('mlab_export.logging.info')
+  def testunit_get_canonical_names_WHEN_experiment_and_skip_metric(
+      self, mock_logging):
+    mock_options = enable_show_options(mock.Mock())
+    mock_options.rrddir_prefix = '/var/lib/collectd/rrd/'
+    fake_filename = ('/var/lib/collectd/rrd/utility.mlab.mlab2.nuq0t/'
+                     'network/if_octets-ipv4.rrd')
+    value_name = 'tx'
+    expected_host = 'mlab2.nuq0t'
+    expected_experiment = 'utility.mlab'
+    expected_metric = None
+    mlab_export.HOSTNAME = 'mlab2.nuq0t'
+    mlab_export.METRIC_MAP = {}
+
+    (returned_host, returned_experiment, returned_metric) = (
+        mlab_export.get_canonical_names(
+            fake_filename, value_name, mock_options))
+    
+    self.assertEqual(returned_host, expected_host)
+    self.assertEqual(returned_experiment, expected_experiment)
+    self.assertEqual(returned_metric, expected_metric)
+    self.assertTrue(mock_logging.called)
+
+  @mock.patch('mlab_export.logging.debug')
+  def testunit_get_json_record(self, mock_logging):
+    timestamps = [1, 2, 3]
+    values = [10.0, 11.0, 12.0]
+    expected_value = {'sample': [
+                       {'timestamp': 1, 'value': 10.0},
+                       {'timestamp': 2, 'value': 11.0},
+                       {'timestamp': 3, 'value': 12.0}],
+                      'metric': 'network.ipv4.bytes.rx',
+                      'hostname': 'mlab2.nuq0t',
+                      'experiment': 'utility.mlab'}
+
+    returned_value = mlab_export.get_json_record(
+        'mlab2.nuq0t', 'utility.mlab', 'network.ipv4.bytes.rx', timestamps,
+        values)
+
+    self.assertEqual(returned_value, expected_value)
+    self.assertTrue(mock_logging.called)
+
+  def testunit_write_json_record(self):
+    output = StringIO.StringIO()
+    record = {'sample': [{'timestamp': 1, 'value': 10.0}],
+              'hostname': 'mlab2.nuq0t',
+              'experiment': 'utility.mlab',
+              'metric': 'network.ipv4.bytes.rx'}
+    pretty_json = None
+    expected_value = ('{"sample": [{"timestamp": 1, "value": 10.0}], '
+                      '"metric": "network.ipv4.bytes.rx", "hostname":'
+                      ' "mlab2.nuq0t", "experiment": "utility.mlab"}\n')
+
+    mlab_export.write_json_record(output, record, pretty_json)
+    returned_value = output.getvalue()
+ 
+    self.assertEqual(returned_value, expected_value)
+
+  def testunit_get_rrd_files(self):
+    rrd_dir = os.path.join(os.path.abspath(self._testdata_dir), 'rrd')
+    expected_files = [rrd_dir + '/mlab2.nuq0t/file2.rrd',
+                      rrd_dir + '/mlab2.nuq0t/file1.rrd']
+
+    returned_files = [ fname for fname in mlab_export.get_rrd_files(rrd_dir) ]
```

Comment:
Good catch. The list comprehension is not necessary here. At one time I was experimenting with using `yield` in get_rrd_files, but that required actually calling the method N times, which the list comprehension did. But, after deciding that `yield` was not really helpful, I forgot to update the test. The test is simpler now: 438fc90cac2709ac217be9175f1a53b20feea305